src/engine/mod.rs

```diff
@@ -564,6 +564,36 @@ where
         Ok(None)
     }
 
+    fn execute_write(&mut self) -> Result<Option<Bug>, EngineError> {
+        if !matches!(self.regs[Register::A0 as usize], Value::Concrete(1)) {
+            return not_supported("can not handle other fd than stdout in write syscall");
+        }
+
+        let buffer = if let Value::Concrete(b) = self.regs[Register::A1 as usize] {
+            b
+        } else {
+            return not_supported(
+                "can not handle symbolic or uninitialized buffer address in write syscall",
+            );
+        };
+
+        let size = if let Value::Concrete(s) = self.regs[Register::A2 as usize] {
+            s
+        } else {
+            return not_supported("can not handle symbolic or uinitialized size in write syscall");
+        };
+
+        trace!("write: fd={} buffer={:#x} size={}", 1, buffer, size,);
+
+        if !self.is_in_vaddr_range(buffer) || !self.is_in_vaddr_range(buffer + size) {
+            return not_supported("write syscall failed to access buffer");
+        }
+
+        self.regs[Register::A0 as usize] = Value::Concrete(size);
```

Comment:
This is a good default implementation I think. The only downside is: by returning the written size, you basically say, that the write syscall is always successful (if inputs are valid). What do you think about returning a symbolic value? In that way we would also explore paths in the user program, where the write syscall "failed". The symbolic value in that case has 3 valid concrete values {0, -1, size}. (https://man7.org/linux/man-pages/man2/write.2.html)

Reply:
Good idea. I am fine extending this to other return values. In fact I think the result can actually take any value in the range [0; size] since partial writes are also possible.

Would you be OK with tackling this extension as part of a separate PR since this PR is getting big as it is? I am a huge fan of smaller incremental changes.

Reply:
yeah, sounds good. Small changes are definitely preferable!